x-pack/plugins/infra/public/alerting/inventory/components/expression_chart.tsx

```diff
@@ -0,0 +1,387 @@
+import { MetricsExplorerRow, MetricsExplorerAggregation } from '../../../../common/http_api';
+import { MetricExplorerSeriesChart } from '../../../pages/metrics/metrics_explorer/components/series_chart';
+import { MetricsExplorerChartType } from '../../../pages/metrics/metrics_explorer/hooks/use_metrics_explorer_options';
+import { getChartTheme } from '../../../pages/metrics/metrics_explorer/components/helpers/get_chart_theme';
+import { calculateDomain } from '../../../pages/metrics/metrics_explorer/components/helpers/calculate_domain';
+import { getMetricId } from '../../../pages/metrics/metrics_explorer/components/helpers/get_metric_id';
+import { useKibanaContextForPlugin } from '../../../hooks/use_kibana';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { InventoryMetricConditions } from '../../../../server/lib/alerting/inventory_metric_threshold/types';
+import { useSnapshot } from '../../../pages/metrics/inventory_view/hooks/use_snaphot';
+import { InventoryItemType, SnapshotMetricType } from '../../../../common/inventory_models/types';
+import { createInventoryMetricFormatter } from '../../../pages/metrics/inventory_view/lib/create_inventory_metric_formatter';
+import { METRIC_FORMATTERS } from '../../../../common/formatters/snapshot_metric_formats';
+import { InfraFormatterType } from '../../../lib/lib';
+
+interface Props {
+  expression: InventoryMetricConditions;
+  filterQuery?: string;
+  nodeType: InventoryItemType;
+  sourceId: string;
+}
+
+const tooltipProps = {
+  headerFormatter: (tooltipValue: TooltipValue) =>
+    moment(tooltipValue.value).format('Y-MM-DD HH:mm:ss.SSS'),
+};
+
+const TIME_LABELS = {
+  s: i18n.translate('xpack.infra.metrics.alerts.timeLabels.seconds', { defaultMessage: 'seconds' }),
+  m: i18n.translate('xpack.infra.metrics.alerts.timeLabels.minutes', { defaultMessage: 'minutes' }),
+  h: i18n.translate('xpack.infra.metrics.alerts.timeLabels.hours', { defaultMessage: 'hours' }),
+  d: i18n.translate('xpack.infra.metrics.alerts.timeLabels.days', { defaultMessage: 'days' }),
+};
+
+export const ExpressionChart: React.FC<Props> = ({
+  expression,
+  filterQuery,
+  nodeType,
+  sourceId,
+}) => {
+  const timerange = useMemo(
+    () => ({
+      interval: `${expression.timeSize || 1}${expression.timeUnit}`,
+      from: moment()
+        .subtract((expression.timeSize || 1) * 20, expression.timeUnit)
+        .valueOf(),
+      to: moment().valueOf(),
+      forceInterval: true,
+      ignoreLookback: true,
+    }),
+    [expression.timeSize, expression.timeUnit]
+  );
+
+  const buildCustomMetric = (metric: any) => ({
+    ...metric,
+    type: 'custom' as SnapshotMetricType,
+  });
+
+  const { loading, nodes } = useSnapshot(
+    filterQuery,
+    expression.metric === 'custom'
+      ? [buildCustomMetric(expression.customMetric)]
+      : [{ type: expression.metric }],
+    [],
+    nodeType,
+    sourceId,
+    0,
+    '',
+    '',
```

Comment:
These empty strings here are the [accountId](https://github.com/elastic/kibana/blob/f8866debf1a36ae5beb2ead5264f3876fde7f978/x-pack/plugins/infra/public/pages/metrics/inventory_view/hooks/use_snaphot.ts#L33) and the [region](https://github.com/elastic/kibana/blob/f8866debf1a36ae5beb2ead5264f3876fde7f978/x-pack/plugins/infra/public/pages/metrics/inventory_view/hooks/use_snaphot.ts#L34) parameters that are sent to the snapshot endpoint.
I see that in other places in the inventory screen we get them with [`useWaffleOptions`](x-pack/plugins/infra/public/pages/metrics/inventory_view/hooks/use_waffle_options.ts) but since users can also create alerts from the Stack Management page I don't think it can be used here, so from where I can get these values?

Reply:
Those values are only available when the user is viewing AWS datasets. To be honest, they probably shouldn't even be there since they act like filters, we should just be adding those as filters to the request instead of having a separate attribute. They are still there for legacy reasons.

Reply:
If I understood it correctly then, for now, I could get these values from `useWaffleOptions` when available. If the alert is created from the Inventory page it would have the correct values, otherwise, it should be ok if they are sent empty if the alert is created from other pages?

Reply:
Yes